backend/email_generator/schemas.py

```diff
@@ -0,0 +1,40 @@
+from typing import Literal
+
+from pydantic import BaseModel, EmailStr, Field
+
+
+class SkillsValidationError(ValueError):
+    """Raised when the skills list is empty or invalid."""
+
+
+class CandidateInfo(BaseModel):
+    """Schema representing candidate information."""
+
+    name: str = Field(..., description="Full name of the candidate.")
+    email: EmailStr = Field(..., description="Email address of the candidate.")
+    skills: list[str] = Field(
+        ..., description="List of candidate's skills (at least one)."
+    )
+    experience: str = Field(..., description="Candidate's experience summary.")
+    title: str = Field(..., description="Job title the candidate applied for.")
+
+
+class EmailGenerationRequest(BaseModel):
+    """Schema representing an email generation request."""
+
+    candidate: CandidateInfo = Field(description="Details of the candidate.")
+    verdict: Literal["Yes", "No"] = Field(
+        ..., description='Final decision: "Yes" for selected, "No" for rejected.'
+    )
+    rejection_reason: str | None = Field(
```

Comment:
Change this to `reason`, since it's possible for HR to add a positive reason for acceptance of a candidate.

```suggestion
    reason: str | None = Field(
```